store/tikv/region_cache.go

```
@@ -246,7 +241,43 @@ func (c *RegionCache) loadRegion(bo *Backoffer, key []byte) (*Region, error) {
 	}
 }
 
-// llrbItem is llrbTree's Item that uses []byte for compare.
+// OnRegionStale removes the old region and inserts new regions into the cache.
+func (c *RegionCache) OnRegionStale(old *Region, newRegions []*metapb.Region) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	c.dropRegionFromCache(old.VerID())
+
+	for _, meta := range newRegions {
+		if err := decodeRegionMetaKey(meta); err != nil {
+			log.Errorf("newRegion's range key is not encoded: %v, %v", meta, err)
```

I think we should return this error.